File: test/codeBase-js/fsm.color/tmp/Green.js
// version: V1

const util = require("util")
const Green = function (context) {
	this.context = context
}

// Partial evaluate
require("./Green.evaluate")(Green)

// ----------
// State Logic Implement Below
// ----------

Green.prototype.OnUpdate = function(stateData, dt) {	
	// Logic Code Here

	return this._evaluate(stateData)
};

Green.prototype.OnEnter = function(stateData) {
	// Logic Code Here
	console.log("Green - Enter")
};

Green.prototype.OnExit = function(stateData) {
	// Logic Code Here
	console.log("Green - Exit")
};

Green.prototype.OnEvent = function(stateData, args) {
	// Logic Code Here
	console.log("Green - On Event ", args)
};

module.exports = Green

File: codeBase/cs/FsmCodeBase/FsmCodeBase/Test/Logic/Green.cs
using System;
using FsmCodeBase;

namespace FsmCodeBase.Test
{
    partial class Green : BaseState
    {
        public override string GetName()
        {
            return "Green";
        }

        public override void OnEnter(StateMachine context)
        {
            Console.WriteLine("[OnEnter] Green");
            base.OnEnter(context);
        }

        public override string OnUpdate(float dt, StateMachine context)
        {
            return base.OnUpdate(dt, context);
        }

        public override void OnEvent(string name, object args, StateMachine context)
        {
            base.OnEvent(name, args, context);
        }

        public override void OnExit(StateMachine context)
        {
            Console.WriteLine("[OnExit] Green");
            base.OnExit(context);
        }
    }
}
